core/backend/src/BackendHubAccess.ts

```diff
@@ -43,6 +43,41 @@ export class LockConflict extends IModelError {
   }
 }
 
+/**
+ * Controls cancellation by raising [[CancelSignal]] on cancel.
+ * @note This is based on AbortController (https://developer.mozilla.org/en-US/docs/Web/API/AbortController).
+ * @beta
+ */
+export class CancelController {
```

Comment:
I'm sorry, I don't understand why any of this is necessary. What's wrong with DownloadProgress returning a bool meaning abort? Way easier to understand and implement for the caller.

Reply:
It's not necessary, but it has several advantages over callback function:

1. Reporting progress may require some extra steps (e.g. querying props of all changesets files to get total size of changesets). This execution time will be wasted if whoever calls download function only cares about cancelling.
1. Using `CancelController` we don't have to wait for progress callback to be called to cancel download.
1. Cancel functionality is much easier to spot when looking at the args of a function rather than going though every arguments documentation (comments) and finding out you have to use progress function to cancel download.

Azure Blob Storage client uses similar args structure ([BlobDownloadOptions interface](https://docs.microsoft.com/en-us/javascript/api/@azure/storage-blob/blobdownloadoptions?view=azure-node-latest)).

If you still think we should go with progress callback, we should reuse [`ProgressCallback`](https://github.com/iTwin/itwinjs-core/blob/9a26879daffa15279d2904c52743b582a807dbbc/core/backend/src/CheckpointManager.ts#L46) (although it returns number rather when boolean).

Reply:
You can change the return type without it being a breaking change due to the way enums work in TypeScript. Please read the documentation, it says:  "If this function returns non-zero, the download is aborted". 

To comply, you should change your code to:
```ts
if (progressFunction() != ProgressStatus.Continue)
``` 

and all is well. Returning "2" aborts the download, as it should.